jbmc/src/java_bytecode/java_entry_point.cpp

```diff
@@ -715,7 +715,7 @@ bool generate_java_start_function(
   symbolt new_symbol;
 
   new_symbol.name=goto_functionst::entry_point();
-  new_symbol.type = code_typet({}, empty_typet());
+  new_symbol.type = java_method_typet({}, empty_typet());
```

Comment:
Empty constructor here

Reply:
Oh OK, my mistake. These are all OK since they relate to a parameters void returning method (so makes sense they have no parameters and no return). 

I guess I would do as you've done here and explicitly call with those two arguments rather than providing the default constructor shortcut. (i.e. get rid of this I think unused constructor https://github.com/diffblue/cbmc/pull/2661/files#diff-2d3ac05ff88eab2a38135f7ff4ef5ad1R251)

Reply:
That sounds like a plan. (BTW, it wasn't my intention to passive aggressively tag these, I was doing it for myself so that we could discuss tomorrow :) )